vnode_controller/vnode_controller.go

```
@@ -443,22 +438,63 @@
-	var err error
-	vn, err := provider.NewVNode(&model.BuildVNodeConfig{
+	vNodeController.createAndRunVNode(initData)
+}
+
+func (vNodeController *VNodeController) createAndRunVNode(initData model.NodeInfo) {
+	nodeName := initData.Metadata.Name
+	vnCtx, vnCtxCancel := context.WithCancel(context.WithValue(context.Background(), "nodeName", nodeName))
+
+	vNode, err := vNodeController.createVNode(vnCtx, initData)
+	if err != nil {
+		err = errpkg.Wrap(err, "Error creating vnode")
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)